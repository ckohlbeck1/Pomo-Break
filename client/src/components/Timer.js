import React from 'react'

const Timer = (props) => {
    const {
        timerLabel,
        clickStartStop,
        startButtonLabel,
        timeLeft,
        clockFormat,
    } = props;

  return (
    <div>
        <h2 id='timer-label'>{timerLabel}</h2> 
        <h3 id='time-left'>{clockFormat(timeLeft)}</h3>
        <button class= "ui button start" id='start-stop' onClick={clickStartStop}>
            {startButtonLabel}
        </button>
    </div>
  )
}

export default Timer
