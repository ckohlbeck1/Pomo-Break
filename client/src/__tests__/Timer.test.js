import React from 'react';
import {render, unmountComponentAtNode} from 'react-dom';
import Timer from '../components/Timer';
import { act } from 'react-dom/test-utils';

it("Timer element renders with text", () => {
    const container = document.createElement("div");
    document.body.appendChild(container);
    act(() => {
        render(<Timer timerLabel="Pomo" timeLeft ="1500" startButtonLabel = "start"/>, container)
    });
    expect(container).toMatchSnapshot();
});