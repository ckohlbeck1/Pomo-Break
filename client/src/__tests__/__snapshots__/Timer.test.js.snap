// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer element renders with text 1`] = `
<div>
  <div>
    <p
      id="timer-label"
    >
      Pomo
    </p>
    1500
    <button
      id="start-stop"
    >
      start
    </button>
  </div>
</div>
`;
